worker_node/job/send.js

```diff
@@ -62,36 +62,49 @@ class Send extends Base {
   }
 
   async initPermission() {
-    const stakingTx = {
-      operation: {
-        type: 'SET_VALUE',
-        ref: `/staking/test/${this.config.ainAddress}/0/stake/${Date.now()}/value`,
-        value: 1,
-        is_global: true,
-      },
-      nonce: -1
-    };
-    const stakingTxResult = await this.#ain.sendTransaction(stakingTx);
-    if (_.get(stakingTxResult, 'result.code') !== 0) {
-      throw Error(`Error while write staking tx (${JSON.stringify(stakingTxResult)})`);
+    const appName = (this.config.transactionOperation.ref).split('/')[2];
+    if (!appName) {
+      throw Error(`Can't find appName from transactionOperation.ref ` +
+          `(${this.config.transactionOperation.ref})`);
+    }
+    const stakingBasePath = `/staking/${appName}/${this.config.ainAddress}/0/stake`;
+    const stakingBaseValue = await this.#ain.db.ref(stakingBasePath).getValue();
+    if (stakingBaseValue === null) {
```

Comment:
Fyi, if you want to check if there's some AIN staked for this app, you could also check if getValue at `/staking/${appName}/balance_total` > 0 !

Reply:
Cool!!!! Done